### PR TITLE
add pluginConfig and implement validator interface

### PR DESCRIPTION
**Pull request by dandans-verily**
Resolve: #4 
Resolve: #5

### CODE REVIEW

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	}, nil
+}
+
+// secretVersion returns the secret data as a string.
+func secretVersion(ctx context.Context, secretVersionName string) (string, error) {
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to set up secret manager client: %w", err)
+	}
+	defer client.Close()
```
Comment (sethvargo) [inline pkg/plugin/plugin.go:97]: This is fine for one secret, but if you end up resolving multiple secrets, you really want to share a client.

File: pkg/plugin/config.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+// PluginConfig defines the set over environment variables required
+// for running the plugin.
+type PluginConfig struct {
+	// JiraEndpoint is the base uri to form the [JIRA REST API uri]. It has the
+	// format of:
+	//     http://host:port/context/rest/api-name/api-version
```
[the remark this answers is not in the document]
Comment (dandans-verily) [reply pkg/plugin/config.go]: Good call!!! Updated to https

File: pkg/plugin/config.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	//
+	// [JIRA REST API url]: https://developer.atlassian.com/server/jira/platform/rest-apis/#uri-structure
+	JiraEndpoint string `env:"JIRA_PLUGIN_ENDPOINT"`
```
[the remark this answers is not in the document]
Comment (dandans-verily) [reply pkg/plugin/config.go]: Updated and updated JIRAAccount accordingly

File: pkg/plugin/config.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		Target:  &cfg.JiraEndpoint,
+		EnvVar:  "JIRA_PLUGIN_ENDPOINT",
+		Example: "https://your-domain.atlassian.net/rest/api/3",
+		Usage: `
+		JiraEndpoint is the base uri to form the [JIRA REST API uri]. It has the
+		format of:
+		    http://host:port/context/rest/api-name/api-version
+		
+		[JIRA REST API url]: https://developer.atlassian.com/server/jira/platform/rest-apis/#uri-structure
+		`,
```
[the remark this answers is not in the document]
Comment (dandans-verily) [reply pkg/plugin/config.go]: Makes sense. Cleaned up throughout.

File: pkg/plugin/config_test.go [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	t.Parallel()
+
+	cases := []struct {
+		name       string
+		envs       map[string]string
+		wantConfig *PluginConfig
+	}{
+		{
+			name: "all_values_specified",
+			envs: map[string]string{
```
[the remark this answers is not in the document]
Comment (dandans-verily) [reply pkg/plugin/config_test.go:39]: Done

File: pkg/plugin/plugin.go (hunk 1 above)
Comment (dandans-verily) [reply pkg/plugin/plugin.go:97]: Yep yep, totally agree!

File: pkg/plugin/config_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (dandans-verily) [reply pkg/plugin/config_test.go:39]: Done

File: pkg/plugin/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dandans-verily) [reply pkg/plugin/config.go]: Right, it was only used in the test, clean it up now after updating the test to use `cli.MapLookuper`

File: pkg/plugin/plugin.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+		validator: v,
+	}, nil
+}
+
+// Validate returns the validation result.
+func (j *JiraPlugin) Validate(ctx context.Context, req *jvspb.ValidateJustificationRequest) (*jvspb.ValidateJustificationResponse, error) {
+	if got, want := req.Justification.Category, jiraCategory; got != want {
+		return nil,
+			fmt.Errorf(
+				"failed to perform validation, expected category %q to be %q",
```
Comment (sethvargo) [inline pkg/plugin/plugin.go:67]: Nit: I'm surprised the linter isn't complaining, but this should be:

```suggestion
			fmt.Errorf("failed to perform validation, expected category %q to be %q",
```

File: pkg/plugin/plugin.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+		return "", fmt.Errorf("failed to set up secret manager client: %w", err)
+	}
+	defer client.Close()
+
+	// Fetch secret version.
+	resp, err := client.AccessSecretVersion(
+		ctx,
+		&secretmanagerpb.AccessSecretVersionRequest{
+			Name: secretVersionName,
+		})
```
Comment (sethvargo) [inline pkg/plugin/plugin.go:104]: Style:

```suggestion
	resp, err := client.AccessSecretVersion(ctx, &secretmanagerpb.AccessSecretVersionRequest{
		Name: secretVersionName,
	})
```

**Review by sethvargo — APPROVED** [review]

**Review by sethvargo — APPROVED** [review]

**Pull request merged**